main.go

```diff
@@ -131,6 +131,7 @@ type WellKnownReport struct {
 
 // A DNSResult is the result of looking up a matrix server in DNS.
 type DNSResult struct {
+	Skipped    bool                  // When the DNS lookup has been skiped due to a port declaration inside the .well-known address.
```

Comment:
I think we should emphasise that this concerns SRV records specifically (since we also do other DNS lookups):

```suggestion
	SRVSkipped bool                  // True if the SRV record lookup was skipped due to the server name or .well-known target including a port number.
```